src/statemachine.rs

```diff
@@ -66,7 +66,84 @@ impl StateMachine {
         Default::default()
     }
 
-    pub fn register(
+    pub fn register<T>(
```

Comment:
You're right, I'm not a big fan of the `_register` / `register<T>` naming convention, but it's a fast rename I think.

My preference would be that the generics-using register/sign, being static methods that take the exact device to work upon, be named `register_dev` and `sign_dev`, so that usage is like 
`StateMachine::register_dev(dev, flags..)`

Whereas the instance methods remain as just `register`/`sign` so they go back to usage of `sm.register(flags..)`

What do you think?